### PR TITLE
Expose set_exponential_reconnect function

### PR DESCRIPTION
**Pull request by MartinHowarth**
Expose the `set_exponential_reconnect` function. Working on testing this at the moment.

This is the default policy (as per the Note in https://docs.datastax.com/en/dev-app-drivers/docs/reconnectionPolicies.html), but exposing it here makes the timings configurable.

Note that the default max delay is actually 600000ms (10 minutes), not 60000ms (1 minute) - the documentation is wrong. The docstring I added for the rust wrapper fixes this error.
c.f. the constant in the C++ driver: https://github.com/datastax/cpp-driver/blob/7f193cb347948b79b1eadf44c8235ee04d66f3cf/src/constants.hpp#L127

### CODE REVIEW

**Comment by angusi** [comment]
Hi @MartinHowarth - please don't forget to read [CONTRIBUTING.md](https://github.com/Metaswitch/cassandra-rs/blob/master/CONTRIBUTING.md) and accordingly sign off your commits with regards the DCO. Thanks!

**Review by kw217 — APPROVED** [review]
LGTM! Thanks @MartinHowarth - but please note Angus's comment; you need to (e-)sign the DCO before this can be merged.

**Comment by MartinHowarth** [comment]
Have signed the DCO and tested it - this is ready to merge if you're happy please. And I'd appreciate a release as well :)

**Pull request merged**
